src/memdb/memdb.ts

```diff
@@ -35,6 +107,50 @@ export const newMemDBPlugin = (): Plugin => {
       const filtered = filterTables(allTables, tables, skipTables, skipDependentTables);
       return await sync(memdbClient, filtered, stream, { deterministicCQId });
     },
+    write(stream: WriteStream): void {
```

Comment:
Should `write` be blocking? I'm wondering if we should change the return value to a promise and resolve it once the stream ended.
Otherwise the call to `write` will return immediately which might be unexpected.

This should also allow us to the reject the promise on stream error

Reply:
You can maybe use https://nodejs.org/api/stream.html#streamfinishedstream-options for this